src/nvim/highlight_group.c

```diff
@@ -2179,6 +2179,7 @@ void highlight_changed(void)
     if (highlight_attr[hlf] != highlight_attr_last[hlf]) {
       if (hlf == HLF_MSG) {
         clear_cmdline = true;
+        update_cmdline = true;
```

Comment:
a new variable in there not be a good choice. tried use `redraw_cmdline` in there seems not works correctly . still use an exist global status variable will be accept I think

Reply:
maybe `need_clr_eos` can be used instead.